routes/nodebb.js

```diff
@@ -159,38 +160,59 @@ app.get(`${BASE_REPORT_URL}/user/uid/:uid`, proxyObject());
 
 function isEditablePost() {
   logger.info({message: "isEditablePost method called"});
-  return function(req, res, next) {
+  return async function(req, res, next) {
     logger.info(req.body);
     const uid = parseInt(req.body.uid || req.query.uid, 10);
     const pid = parseInt(req.params.pid, 10);
-    const url = `${req.protocol}://${req.get('host')}${BASE_REPORT_URL}/post/pid/${pid}`
-    const options = {
+    const url = `${req.protocol}://${req.get('host')}${BASE_REPORT_URL}/v3/posts/${pid}`
+    let options = {
       url: url,
       method: 'GET',
       json: true
     };
     logger.info(options)
+    let response;
+    try {
+      response = await getPostDetails(options);
+    } catch(error) {
+        if (error.statusCode === 404) {
+          const url = `${req.protocol}://${req.get('host')}${BASE_REPORT_URL}/post/pid/${pid}`;
+          options.url = url;
+          response = await getPostDetails(options);
+        } else {
+          next(error);
+        }
+    }
+
+    logger.info(response)
+    if (response.uid === uid && response.pid === pid) {
+      logger.info({message: 'Uid got matched and the post can be deleted'})
+      next();
+    } else {
+      logger.info({message: 'Uid is not matched and you can not delete the post'})
+      res.status(400)
+      res.send(responseObj)
+    }
+  }
+}
+
+function getPostDetails(options) {
+  if (options) {
+    return new Promise((resolve, reject) => {
       request(options, (error, response, body) => {
-        if(error) {
+        if(error || response.statusCode === 404) {
           logger.info({message: `Error while call the api ${options.url}`})
-          logger.info({message: `Error message:  ${error.message}`})
-          next(error);
+          logger.info({message: `Error message:  ${error}`})
```

Comment:
Put the logger saying old nodeV.1.16 is used. So that we can Identify how many adopters are still using 1.16 later. Helps to remove the code if no logs exist with 1.16